Rename datetime_to_index to to_internal_index

cwasicki opened the pull request:
The method is renamed to avoid confusion between indices in a window, which are passed by the user, and the internal indices that point to the actual position in the internal buffer.

cwasicki committed:
Rename datetime_to_index to to_internal_index

The method is renamed to avoid confusion between indices in a window,
which are passed by the user, and the internal indices that point to the
actual position in the internal buffer.

Signed-off-by: cwasicki <126617870+cwasicki@users.noreply.github.com>

llucax reviewed (dismissed):
LGTM. Approving but tagging @mathias-baumann-frequenz and @Marenz just in case they have something to say.

cwasicki commented:
Rebased on latest v0.x.x, signed and updated release notes (merge with changes from https://github.com/frequenz-floss/frequenz-sdk-python/pull/670). Everything ok from your side @Marenz @matthias-wende-frequenz ?
